File: w1/home/square2rev/square2rev.c
#include <stdio.h>

int main() {
    int size, start, end;
    
    scanf("%d", &size);
    
    end = size * size;
    start = end - size + 1;
    
    for ( int row = 1; row <= size; row++ ) {
        for ( int row = 1; row < size; row++ ) {
            printf("%d ", start);
            start += 1;
        }
        printf("%d\n", end);
        end -= size;
        start = end - size + 1;
    }
    
    return 0;
}

File: w4/php_dz/square2rev.php
<?php
    $size = rtrim(fgets(STDIN));
    $out = "";

    for ( $row = 1, $end = $size * $size, $start = $end - $size + 1; $row <= $size; $row++ ) {
        for ( $col = 1; $col < $size; $col++ ) {
            $out .= $start . " ";
            $start += 1;
        }
        $out .= $end . PHP_EOL;
        $end -= $size;
        $start = $end - $size + 1;
    }

    echo $out;
?>